src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/StripEmbeddedLibraries.cs

```diff
@@ -4,9 +4,192 @@
+			designerConstants = BuildResourceDesignerFieldLookup (mainDesigner);
+		}
+
+		protected override void EndProcess ()
+		{
+			if (mainDesigner != null) {
+				RemoveDesigner (mainDesigner);
+				Annotations.SetAction (mainAssembly, AssemblyAction.Save);
+			}
+		}
+
+		bool FindResourceDesigner (AssemblyDefinition assembly, bool isApplication, out TypeDefinition designer)
+		{
+			string designerFullName = null;
+			designer = null;
+			foreach (CustomAttribute attribute in assembly.CustomAttributes)
+			{
+				if (attribute.AttributeType.FullName == "Android.Runtime.ResourceDesignerAttribute")
+				{
+					if (attribute.HasProperties)
+					{
+						foreach (var p in attribute.Properties)
+						{
+							if (p.Name == "IsApplication" && (bool)p.Argument.Value == isApplication)
+							{
+								designerFullName = attribute.ConstructorArguments[0].Value.ToString();
+								break;
+							}
+						}
+					}
+					break;
+
+				}
+			}
+			if (string.IsNullOrEmpty(designerFullName))
+				return false;
+			foreach (ModuleDefinition module in assembly.Modules)
+			{
+				foreach (TypeDefinition type in module.Types)
+				{
+					if (type.FullName == designerFullName)
+					{
+						designer = type;
+						return true;
+					}
+				}
+			}
+			return false;
+		}
+
+		Dictionary<string, int> BuildResourceDesignerFieldLookup(TypeDefinition type)
+		{
+			var output = new Dictionary<string, int> ();
+			foreach (TypeDefinition definition in type.NestedTypes)
+			{
+				foreach (FieldDefinition field in definition.Fields)
+				{
+					string key = $"{definition.Name}::{field.Name}";
+					if (!output.ContainsKey (key))
+						output.Add(key, int.Parse (field.Constant?.ToString () ?? "0"));
+				}
+			}
+			return output;
+		}
+
+		void RemoveDesigner(TypeDefinition designer)
+		{
+			Context.LogMessage ($"    TryRemoving {designer.Name}");
+			var module = designer.Module;
+			if (module != null) {
+				Context.LogMessage ($"    Removed {designer.Name}");
+				module.Types.Remove(designer);
+			}
+		}
+
+		void FixBody (MethodBody body, TypeDefinition localDesigner)
+		{
+			Dictionary<Instruction, int> instructions = new Dictionary<Instruction, int>();
+			var processor = body.GetILProcessor();
+			string designerFullName = $"{localDesigner.FullName}/";
+			foreach (var i in body.Instructions)
+			{
+				string line = i.ToString ();
+				if (line.Contains (designerFullName) && !instructions.ContainsKey (i))
+				{
+					var match = opCodeRegex.Match (line);
+					if (match.Success && match.Groups.Count == 5) {
+						string key = match.Groups[4].Value.Replace (designerFullName, string.Empty);
+						if (designerConstants.ContainsKey (key))
+							instructions.Add(i, designerConstants[key]);
+					}
+				}
+			}
+			if (instructions.Count > 0)
+				Context.LogMessage ($"    Fixing up {body.Method.FullName}");
+			foreach (var i in instructions)
+			{
+				var newCode = processor.Create(OpCodes.Ldc_I4, i.Value);
```

Comment:
This can be optimized a bit by using [CreateLoadArraySizeOrOffsetInstruction](https://github.com/xamarin/xamarin-android/blob/d471b4a4308310afb8fa3fa1047003a7caedd57f/src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MonoDroidMarkStep.cs#L318) plus maybe some refactoring to move it outside of `MonoDroidMarkStep`, so that we can build it easily for .NET5/6.

Reply:
@radekdoulik not sure I understand what you mean by "move it outside of MonoDroidMarkStep" ? can you explain in more detail?

Reply:
The `MonoDroidMarkStep` is not built on .NET5/6, so if we would like to use the code in step used in .NET5/6, we need to move it to a helper class or some other place.